extension/src/popup/ducks/transactionSubmission.ts

```diff
@@ -261,20 +245,128 @@ const storeBalanceMetricData = (publicKey: string, accountFunded: boolean) => {
   localStorage.setItem(METRICS_DATA, JSON.stringify(metricsData));
 };
 
-export const getAccountBalances = createAsyncThunk<
-  AccountBalancesInterface,
-  { publicKey: string; networkDetails: NetworkDetails },
+export const removeTokenId = createAsyncThunk<
+  void,
+  {
+    contractId: string;
+    network: NETWORKS;
+    sorobanClient: SorobanContextInterface;
+  },
   { rejectValue: ErrorMessage }
->("getAccountBalances", async ({ publicKey, networkDetails }, thunkApi) => {
+  // @ts-ignore
+>("removeTokenId", async ({ contractId, network, sorobanClient }, thunkApi) => {
   try {
-    const res = await internalGetAccountBalances({ publicKey, networkDetails });
-    storeBalanceMetricData(publicKey, res.isFunded || false);
-    return res;
+    await internalRemoveTokenId({ contractId, network });
   } catch (e) {
-    return thunkApi.rejectWithValue({ errorMessage: e });
+    console.error(e);
+    thunkApi.rejectWithValue({ errorMessage: e as string });
   }
 });
 
+export const getAccountBalancesWithFallback = createAsyncThunk<
```

Comment:
There's a lot of logic here but it's the one place where we house the complexity of the fallback strategy, should this be broken up/split up in some way?